LanguageFeatures/Inline-classes/static_analysis_inline_class_A12_t04.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let DV be an inline class declaration named Inline with type
+/// parameters X1 extends B1, .. Xs extends Bs. Assume that DV declares a final
+/// instance variable with name id and type R.
+///
+/// We say that the declared representation type of Inline is R, and the
+/// instantiated representation type corresponding to Inline<T1,.. Ts> is
+/// [T1/X1, .. Ts/Xs]R.
+/// ...
+/// Let V be an inline type of the form Inline<T1, .. Ts>, and let R be the
+/// corresponding instantiated representation type. If R is non-nullable then V
+/// is a proper subtype of Object, and V is non-nullable. Otherwise, V is a
+/// proper subtype of Object?, and V is potentially nullable.
+///
+/// @description Checks that if an instantiated representation type `R` is
+/// not non-nullable then an inline type `V` is potentially nullable (it's a
+/// compile-time error to assign `Object?` or `null` to `V`)
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+inline class V1 {
+  final int? it;
+  V1(this.it);
+}
+
+inline class V2<T1, T2 extends num?> {
+  final T1? id;
```

Comment:
The test would be stronger if this were `T1` rather than `T1?`, and then ..

Line 46: Declared type `V2<String?, int>` would have a nullable representation type.
Line 51: The raw `V2` as the declared type would also have a nullable representation type, but it's probably better to use `V2<String?, int>` again, just so we're testing `= anObjectQuestion` and `= null` on the same declared type, following the pattern set by line 36 and 41.